pkg/relay/relay.go

```diff
@@ -0,0 +1,513 @@
+//go:generate mockgen  -destination=./mocks/mocks.go -package=mocks github.com/blocknative/dreamboat/pkg/relay Datastore,State
+package relay
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/flashbots/go-boost-utils/bls"
+	"github.com/flashbots/go-boost-utils/types"
+	"github.com/lthibault/log"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/blocknative/dreamboat/pkg/structs"
+)
+
+type State interface {
+	Beacon() *structs.BeaconState
+}
+
+var (
+	ErrNoPayloadFound        = errors.New("no payload found")
+	ErrMissingRequest        = errors.New("req is nil")
+	ErrMissingSecretKey      = errors.New("secret key is nil")
+	UnregisteredValidatorMsg = "unregistered validator"
+	noBuilderBidMsg          = "no builder bid"
+	badHeaderMsg             = "invalid block header from datastore"
+)
+
+type Datastore interface {
+	PutDelivered(context.Context, structs.Slot, structs.DeliveredTrace, time.Duration) error
+	GetDelivered(context.Context, structs.Query) (structs.BidTraceWithTimestamp, error)
+
+	PutPayload(context.Context, structs.PayloadKey, *structs.BlockBidAndTrace, time.Duration) error
+	GetPayload(context.Context, structs.PayloadKey) (*structs.BlockBidAndTrace, error)
+
+	PutHeader(context.Context, structs.Slot, structs.HeaderAndTrace, time.Duration) error
+	GetHeaders(context.Context, structs.Query) ([]structs.HeaderAndTrace, error)
+	GetMaxProfitHeadersDesc(context.Context, structs.Slot) ([]structs.HeaderAndTrace, error)
+
+	PutRegistrationRaw(context.Context, structs.PubKey, []byte, time.Duration) error
+	GetRegistration(context.Context, structs.PubKey) (types.SignedValidatorRegistration, error)
+}
+
+type RegistrationManager interface {
```

Comment:
This interface is hard to understand. I think we should change the method names. Since in order to understand what they do, I must go to the implementation

Reply:
Nice!